src/main/bootstrap/cache.rs

```diff
@@ -82,25 +74,15 @@ impl Cache {
     }
 
     /// Inserts given peer to the cache. If peer is already cached, it is moved to cache front.
-    ///
-    /// ## Returns
-    ///
-    /// A list of expired peers.
-    pub fn put(&mut self, peer: PeerInfo) -> HashSet<PeerInfo> {
-        let (_, mut expired) = self.peers.notify_insert(peer.pub_key, peer);
-        expired.drain(..).map(|(_key, val)| val).collect()
-    }
-
-    /// Removes given peer from the cache.
-    pub fn remove(&mut self, peer: &PeerInfo) {
```

Comment:
I think we should still have the remove API even if we don't use it just now. I am thinking about cases where we _know_ that a peer is malicious. In that case i don't think it makes sense to keep them in the cache (if we are blacklisting them or something) even if it mean decrementing the peer-count in cache.